kernel/src/snapshot.rs

```diff
@@ -225,9 +124,7 @@ impl Snapshot {
         version: Version,
         engine: &dyn Engine,
     ) -> DeltaResult<Self> {
-        let (metadata, protocol) = log_segment
-            .read_metadata(engine)?
-            .ok_or(Error::MissingMetadata)?;
+        let (metadata, protocol) = log_segment.read_metadata(engine)?;
```

Comment:
did we change read_metadata?

Reply:
Yes, it used to return `DeltaResult<Option<(Metadata, Protocol)>>`, but the function never returned `Ok(None)`. If there's no `Protocol` or no `Metadata`, this is an error.

Reply:
cool sgtm!